webhook/admission.go

```diff
@@ -94,17 +96,24 @@ func admissionHandler(wh *Webhook, c AdmissionController, synced <-chan struct{}
 		logger := wh.Logger
 		logger.Infof("Webhook ServeHTTP request=%#v", r)
 
+		span := trace.SpanFromContext(r.Context())
+		// otelhttp middleware creates the labeler
+		labeler, _ := otelhttp.LabelerFromContext(r.Context())
+
+		defer func() {
+			span.SetAttributes(labeler.Get()...)
+		}()
+
 		var review admissionv1.AdmissionReview
 		bodyBuffer := bytes.Buffer{}
 		if err := json.NewDecoder(io.TeeReader(r.Body, &bodyBuffer)).Decode(&review); err != nil {
-			http.Error(w, fmt.Sprint("could not decode body:", err), http.StatusBadRequest)
+			err := fmt.Sprint("could not decode body:", err)
+			span.SetStatus(codes.Error, err)
```

Comment:
One thing to note is that `otelhttp` doesn't mark 4xx status as errors on the span.

I'm unsure if that's a recommendation on what's an error or not. So I might be inclined to set the status to these as errors

Reply:
my preference would also be to set these as errors